storage-proofs/src/stacked/proof.rs

```diff
@@ -487,50 +490,33 @@ impl<'a, H: 'static + Hasher, G: 'static + Hasher> StackedDrg<'a, H, G> {
         // offset into the first window
         let first_window_offset = offset % pp.window_size_bytes();
 
-        // length of the last window
-        let last_window_len = if (num_bytes + first_window_offset) % pp.window_size_bytes() > 0 {
-            (num_bytes + first_window_offset) % pp.window_size_bytes()
-        } else {
-            pp.window_size_bytes()
-        };
-
         // determine the number of windows need to be decoded
-        let num_windows = (num_bytes as f64 / pp.window_size_bytes() as f64).ceil() as usize;
+        let num_windows =
+            ((offset + num_bytes) as f64 / pp.window_size_bytes() as f64).ceil() as usize;
```

Comment:
What if `num_bytes` is very small and `offset` is large. Won't `num_windows` be large, when it should be `1`?

I think what you are calling `num_windows` is actually `last_window_index`, and `num_window = last_window_index - first_window_index` or something like that.

Reply:
I need to update the comment, and maybe rename the variable. If you look how I use it, I first take `num_windows` and then skip `first_window_index`. It now is more like `last_window_index`

Reply:
Got it. I do recommend making those clarification so correctness is obvious to future readers. But looks like this is in order then.